### PR TITLE
Expose trigger enabled state

### PR DESCRIPTION
**Pull request by christophermaier**

### CODE REVIEW

File: lib/cog_api/resources/trigger.ex [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
   @derive [Poison.Encoder]
 
-  defstruct [:id,
-             :name,
-             :pipeline,
-             :as_user,
-             :timeout_sec,
-             :description,
-             :invocation_url]
+  defstruct [id: nil,
```
Comment (drapergeek) [inline lib/cog_api/resources/trigger.ex]: What's the reasoning on moving all these to defaults?

Comment (christophermaier) [inline lib/cog_api/resources/trigger.ex]: @drapergeek I added the `enabled` field, which defaults to true, and I can't mix bare atoms and Keywords.

Comment (drapergeek) [inline lib/cog_api/resources/trigger.ex]: You can move the keywords to the bottom of the list and leave the others as atoms. I don't have a strong opinion on it just seems weird to explicitly define things as `nil`.

Comment (christophermaier) [inline lib/cog_api/resources/trigger.ex]: Figures: that was the one thing I didn't try 😄 

TIL

**Comment by drapergeek** [comment]
:shipit:

**Pull request merged**